tests/features/plugins.spec.ts

```diff
@@ -0,0 +1,67 @@
+import { mount, VueWrapper, config } from '../../src'
+
+declare module '../../src/vue-wrapper' {
+  interface VueWrapper {
+    width(): number
+    $el: Element
+    myMethod(): void
+  }
+}
+
+const textValue = `I'm the innerHTML`
+const mountComponent = () => mount({ template: `<h1>${textValue}</h1>` })
+
+describe('Plugin', () => {
+  describe('#install method', () => {
+    beforeEach(() => {
+      config.plugins.VueWrapper.reset()
+    })
+
+    it('extends wrappers with the return values from the install function', () => {
+      const width = 230
+      const plugin = () => ({ width })
+      config.plugins.VueWrapper.install(plugin)
+      const wrapper = mountComponent()
+      expect(wrapper).toHaveProperty('width', width)
+    })
+
+    it('receives the wrapper inside the plugin setup', () => {
+      const plugin = (wrapper) => {
+        return {
+          $el: wrapper.element // simple aliases
```

Comment:
Will we have access to private properties on the instance? Like the rootVM?

Reply:
Yeah, your original example will work. You can get access to the `ComponentPublicInstance` under `componentVM` which is private.

Reply:
```js
const namePlugin = wrapper => ({ lowerCaseName: wrapper.componentVM.$.type.name })
config.plugins.VueWrapper.install(namePlugin)
const wrapper = mount({ template: `<h1>Hello</h1>`, name: 'My_Component' })
wrapper.lowerCaseName() // 'my_component'
```